src/python_testing/TC_DISHM_2_1.py

```diff
@@ -0,0 +1,283 @@
+        supported_modes_dut = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=supported_modes_attribute)
+        modes = [m.mode for m in supported_modes_dut]
+
+        logging.info(f"SupportedModes: {supported_modes_dut}")
+        logging.info(f"Modes: {modes}")
+
+        can_test_mode_failure = self.check_pics("DISHM.S.M.CAN_TEST_MODE_FAILURE")
+        can_manually_control = self.check_pics("DISHM.S.M.CAN_MANUALLY_CONTROLLED")
+
+        # Check if the list of supported modes is larger than 2
+        asserts.assert_greater_equal(len(supported_modes_dut), 2, "SupportedModes must have at least 2 entries!")
+
+        if can_test_mode_failure and can_manually_control:
+            # Verify that PIXIT.DISHM.MODE_CHANGE_FAIL is one of supported_modes_dut
+            asserts.assert_true(self.mode_fail in modes,
+                                f"{self.mode_fail} is not in supported_modes_dut: {modes}")
+
+        # TH reads from the DUT the CurrentMode attribute
+        self.step(3)
+        old_current_mode_dut = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=current_mode_attribute)
+
+        # CurrentMode attribute value is an integer value
+        is_valid_int_value(old_current_mode_dut)
+        asserts.assert_in(old_current_mode_dut, modes, "Current mode is not in the supported modes")
+
+        invalid_mode_th = max(modes) + 1
+        asserts.assert_true(invalid_mode_th not in modes,
+                            f"{invalid_mode_th} is in {modes} and it shouldn't")
+
+        # TH sends a ChangeToMode command to the DUT with NewMode set to old_current_mode_dut
+        self.step(4)
+        cmd = cluster.Commands.ChangeToMode(newMode=old_current_mode_dut)
+        change_to_mode_response = await self.send_single_cmd(cmd=cmd, endpoint=endpoint)
+
+        # DUT responds contains a ChangeToModeResponse command with a SUCCESS (value 0x00) status response
+        asserts.assert_true(matchers.is_type(change_to_mode_response, cluster.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        asserts.assert_equal(change_to_mode_response.status, CommonCodes.SUCCESS.value,
+                             f"Status is {change_to_mode_response.status} and it should be SUCCESS 0x00")
+
+        # Manually put the device in a state from which it will FAIL to transition to PIXIT.DISHM.MODE_CHANGE_FAIL
+
+        # This step prepares the DUT for a negative transition scenario.
+        # The goal is to place the DUT in a state where a subsequent ChangeToMode(newMode=MODE_CHANGE_FAIL)
+        # would fail (e.g. device is busy/incompatible state). On real devices this can be done
+        # manually, in CI the example app does not expose a way to synthesize the internal condition.
+
+        if not can_test_mode_failure:
+            self.skip_step(5)
+        elif not can_manually_control:
```

Comment:
The above should be guarded on both can_test_mode_failure and can_manually_control. The write to app pipe should be guarded on this being CI. CI should have the manually controlled PICS set - I know it's not actually manually controlled, but that's what the app pipe is mocking.